proposals/0065-issue-pr-label-system.md

```diff
@@ -0,0 +1,254 @@
+
+Component labels indicate which part of the codebase an issue or PR affect.
+Area labels provide additional categorization for cross-cutting concerns.
+A few examples of currently used labels:
+
+- **`component/api`**: HTTP API
+- **`component/promql`**: PromQL query engine
+- **`component/promtool`**: promtool CLI utility
+- **`component/rules`**: Recording and alerting rules
+- **`component/scraping`**: Metric scraping and service discovery
+- **`component/service discovery`**: Service discovery mechanisms
+- **`component/tsdb`**: Time series database (storage engine)
+- **`component/ui`**: Web UI
+- **`area/build`**: Build system and build process
+- **`area/ci-cd`**: Continuous integration and deployment
+- **`area/opentelemetry`**: OpenTelemetry integration and compatibility
+- **`area/utf8`**: UTF-8 support and related issues
+
+**Usage**: Issues and PRs typically have one or more component labels to
+indicate the affected areas. Multiple component labels may be applied if changes
+span multiple components. Area labels are less commonly used than component
+labels and typically indicate cross-cutting concerns or specific initiatives.
+
+#### Kind Labels
+
+Kind labels categorize the type of change or issue:
+
+- **`kind/bug`**: Something is broken or not working as intended
+- **`kind/enhancement`**: Improvement to existing functionality
+- **`kind/feature`**: Entirely new functionality
+- **`kind/cleanup`**: Code cleanup, refactoring, or technical debt reduction
+- **`kind/optimization`**: Performance improvements
+- **`kind/change`**: General change that doesn't fit other categories
+- **`kind/breaking`**: Breaking change that affects backward compatibility
+
+**Usage**: Issues and PRs typically have exactly one kind label to indicate the primary nature of the change. The kind label helps communicate the impact and urgency of the change.
+
+#### Priority Labels
+
+Priority labels indicate the urgency and importance of an issue or PR:
+
+- **`priority/P0`**: Critical priority - requires immediate attention (e.g., production outages, security issues)
+- **`priority/P1`**: High priority - should be addressed in the current release cycle
+- **`priority/P2`**: Medium priority - should be addressed soon but not urgent
+- **`priority/P3`**: Low priority - nice to have but can wait
+- **`priority/Pmaybe`**: Lowest priority - may or may not be addressed; needs further discussion
+
+**Usage**: Priority labels are typically assigned during triage and help maintainers and contributors understand what to work on first. Not all issues have priority labels assigned.
+
+#### Other Common Labels
+
+Many other labels are currently in use. this proposal does not seek to prescribe
+any changes to how all labels are used. For example to following are well
+established and usefule:
+
+- **`help wanted`**: Good issues for external contributors
+- **`good first issue`**: Good for newcomers to the project
+- **`low hanging fruit`**: Easy to implement
```

Comment:
It always rubbed me the wrong way that we have these two labels. Their difference is very subtle in practice.